osvr/RenderKit/RenderManagerD3D11ATW.h

```diff
@@ -489,6 +517,21 @@ namespace osvr {
                 std::vector<osvr::renderkit::RenderBuffer> renderBuffers;
                 size_t numRenderInfos = renderInfo.size();
 
+                if (renderInfo.size() == 0) {
+                    m_log->error()
+                        << "RenderManagerD3D11ATW::"
+                        << "RegisterRenderBuffersInternal: renderInfo was zero length.";
+                    return false;
+                }
+
+                // @todo: we need to get the actual device for the given render info associated
+                // with the given buffer being registered. Since we support re-using the buffers
+                // for more than one eye, then we need to actually track this somehow, but we
+                // can't with the given parameters without just making assumptions
+                // for now, just use the first eye's device, but this may not work with
+                // multi-display HMDs
+                auto atwDevice = renderInfo[0].library.D3D11->device;;
```

Comment:
The problem is that we can't match RenderInfo index to buffer when they could either be double buffering or single buffering AND/OR they could be combining multiple renderInfo into one buffer. 

 1. If they are single buffering, using one buffer per eye, then buffer 0 goes with render info 0, buffer 1 goes with render info 1. The i % 2 trick works.
 1. If they are single buffering, but using one buffer for both eyes, then there is only one buffer and it goes with both render info 0 and 1. The i % 2 trick does NOT work.
 1. If they are double buffering, but using one buffer for both eyes, then buffer 0 goes with both render info 0 and 1. As does buffer 1. The i % 2 trick does NOT work.
 1. If they are double buffering with one buffer per eye, there will be 4 buffers. Buffer 0 goes with render info 0, buffer 1 goes with render info 1, buffer 2 goes with render info 0, and buffer 3 goes with render info 1. The i % 2 trick works.

The API does not have a way to specify how registered buffers map to a given render info. The client could register 1, 2, or 4 buffers (1, 2, 3, 4, or 6 if you allow triple buffering as an option), which map to 2 render infos in different ways depending on how they intend to render into the buffers (both eyes in one texture or one texture per eye).